todo-app/src/todo/controller.ts

```diff
@@ -0,0 +1,54 @@
+import {
+    Body,
+    Controller,
+    Delete,
+    Get,
+    Param,
+    ParseIntPipe,
+    Post,
+    Put,
+  } from '@nestjs/common';
+  import { TodoService } from './service';
+  import {
+    CreateTodoRequestDto,
+    GetTodoDateRequestDto,
+  } from './dto/request.dtos';
+  import { TodoResponseDto } from './dto/response.dtos';
+  
+  @Controller('todo')
+  export class TodoController {
+    constructor(private todoService: TodoService) {}
+  
+    // create todo
+    @Post()
+    createTodo(
+      @Body() requestDto: CreateTodoRequestDto,
+    ): Promise<TodoResponseDto> {
+      return this.todoService.createTodo(requestDto);
+    }
+  
+    // get all todos for the username̦
+  
+    @Get('/:userName')
+    getTodoByUsername(
+      @Param('userName') userName: string,
+    ): Promise<TodoResponseDto[]> {
```

Comment:
- by default, always avoid sending an array directly as a response
- if you want to send this as a response `[ { id: 2, name: "ABCD" }, {id: 3, name: "PQR" } ]`
- then send it like this `{ list: [ { id: 2, name: "ABCD" }, {id: 3, name: "PQR" } ] }`
- this allows us to later add more info to the response without dramatically changing the structure
- for example: `{ list: [ { id: 2, name: "ABCD" }, {id: 3, name: "PQR" } ], updatedAt: "2024" }`